sklearn_cotraining/classifiers.py

```diff
@@ -128,11 +127,7 @@ def fit(self, X1, X2, y):
 			L.extend([U_[x] for x in p])
 			L.extend([U_[x] for x in n])
 
-			#TODO: optimize these removals from U_
-			#this is currently (2p + 2n)O(n)
-			#and I think it can be reduced to O(n) rather easily
-			for i in p: U_.pop(i)
-			for i in n: U_.pop(i)
+			U_ = [elem for elem in U_ if (elem in p or elem in n)]
 
 			#add new elements to U_
 			add_counter = 0 #number we have added from U to U_
```

Comment:
Do we still need this block here to refill U_?

Reply:
We are still using the old code to refill U_.